### DISCUSSION TITLE
Adding subscribers to AGW in headless mode

### DISCUSSION THREAD

**Discussion by ulaskozat** [magma/magma#7522] [Show and tell]
A lot of times folks are coming and asking about how to use AGW in a headless manner without using Orc8r. In most cases, people want to simply test the core network and understand the call flows.  In such cases, you need to make sure that eNB configuration and subscriber configurations are done properly. If for instance eNB tracking area code does not match to any of the TACs in TAI list on MME side, the s1 setup would fail. If subscriber is not added properly to the subscriberdb, then attach requests would be rejected with cause values depending on the missing configuration. **Below is how you can manage subscribers in headless mode, i.e., AGW is not registered with and configured by any Orc8r**. Otherwise all your subscriber configs will be overwritten by the configs done on the cloud end.

First add the subscriber:
`subscriber_cli.py add --lte-auth-key <hexadecimal_lte_key> <imsi>`
e.g., `subscriber_cli.py add --lte-auth-key 8AAC473F2F8FD09987CCCBD7097C6963 IMSI001010000324582`

Then add an apn using the update subcommand:
`subscriber_cli.py update --apn-config internet,9,1,0,0,3000,4000,0,,,,`

Note that we have some consecutive commas such as`,,,`, that simply means we left some optional configs untouched. To get a full understanding of all the update options and how to use `--apn-config`, simply execute:
`subscriber_cli.py update -h`. If you are on dev vm, you need to first enter your virtual env via executing command `magtivate`.

**Comment by pramchan** [comment]
LGTM for manual testing for standalone AGW.
